httpclient/src/main/java/org/apache/http/impl/client/HttpClientBuilder.java

```diff
@@ -954,7 +996,18 @@ public CloseableHttpClient build() {
             requestExecCopy = new HttpRequestExecutor();
         }
         HttpClientConnectionManager connManagerCopy = this.connManager;
-        if (connManagerCopy == null) {
+
+        final String socksProxyHost = System.getProperty("socksProxyHost");
+        final String socksProxyPort = System.getProperty("socksProxyPort");
+
+        if (systemProperties && socksProxyHost != null && socksProxyPort != null) {
+            // TODO should use a method that creates a new connection manager based on the current, but with a SOCKS registry
+            // replaces the current connection manager with a SOCKS oriented connection manager with configuration based on this HttpClient.
+            connManagerCopy = createConnectionManager(RegistryBuilder.<ConnectionSocketFactory>create()
+                    .register("http", SOCKSConnectionSocketFactory.getSocketFactory())
+                    .register("https", SOCKSConnectionSocketFactory.getSocketFactory())
```

Comment:
This should be an SSL based socket. This pull request needs some more effort. With tests.